source/common/common/envoy_defaults.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include "source/common/common/manifest.pb.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/singleton/threadsafe_singleton.h"
+
+// These macros return the value associated with `field_name` in the protobuf `message` if it
+// exists. Otherwise, they return the value associated with `field_name` in the defaults profile if
+// it exists. Otherwise, `default_value` is returned.
+
+#define PROTOBUF_GET_NUMBER_OR_PROFILE_DEFAULT(message, config, field_name, default_value)         \
+  ([&](const auto& msg) -> double {                                                                \
+    if (msg.has_##field_name()) {                                                                  \
+      return msg.field_name().value();                                                             \
+    }                                                                                              \
+    return DefaultsProfile::get().getNumber(config, #field_name, default_value);                   \
+  }((message)))
+
+#define PROTOBUF_GET_MS_OR_PROFILE_DEFAULT(message, config, field_name, default_value)             \
+  ([&](const auto& msg) -> std::chrono::milliseconds {                                             \
```

Comment:
Please don't use `&`, it captures way too much. What do you need to capture here?

Reply:
This is a good point. I played with this lambda capture a lot, and unfortunately the only way I got it to compile was using `[&]`. I think I need to capture `config`, but when I write `[&config]`, or `[config]`, or add `config` as an argument to the lambda like `msg`, it doesn't build. I would figure this issue out instead of leaving this comment, but today's the last day of my internship so hopefully Yan or Adi can take a look.